lib/ueberauth/strategy/slack.ex

```diff
@@ -189,18 +189,24 @@ defmodule Ueberauth.Strategy.Slack do
   # Given the auth and token we can now fetch the user.
   defp fetch_user(conn, token) do
     auth = conn.private.slack_auth
+    scope_string = (token.other_params["scope"] || "")
+    scopes       = String.split(scope_string, ",")
 
-    case OAuth2.AccessToken.post(token, "/users.info", [token: token.access_token, user: auth["user_id"]], [{"Content-Type", "application/x-www-form-urlencoded"}]) do
-      {:ok, %OAuth2.Response{status_code: 401, body: _body}} ->
-        set_errors!(conn, [error("token", "unauthorized")])
-      {:ok, %OAuth2.Response{status_code: status_code, body: user}} when status_code in 200..399 ->
-        if user["ok"] do
-          put_private(conn, :slack_user, user["user"])
-        else
-          set_errors!(conn, [error(user["error"], user["error"])])
+    case "users:read" in scopes do
+      false -> conn
```

Comment:
Do we need to set an error here? I'm not sure how the rest of the module would react to not having a slack user on the conn when it tries to construct the auth struct.